ip/flac.c

```diff
@@ -427,11 +415,14 @@ static int flac_seek(struct input_plugin_data *ip_data, double offset)
 
 	sample = (uint64_t)(offset * (double)sf_get_rate(ip_data->sf) + 0.5);
 	if (!F(seek_absolute)(priv->dec, sample)) {
+		if (F(get_state(priv->dec)) == FLAC__STREAM_DECODER_SEEK_ERROR) {
+			if (!F(flush)(priv->dec))
+				d_print("failed to flush\n");
+			priv->buf_rpos = 0;
+			priv->buf_wpos = 0;
+		}
 		return -IP_ERROR_ERRNO;
 	}
-	priv->ignore_next_write = 1;
-	priv->buf_rpos = 0;
-	priv->buf_wpos = 0;
```

Comment:
the FLAC__stream_decoder_seek_absolute  will internally call seek_to_absolute_sample_, which process a frame. See https://github.com/xiph/flac/blob/4f47b63e9c971e6391590caf00a0f2a5ed612e67/src/libFLAC/stream_decoder.c#L3140 . 

so the wirte_cb will be called during flac_seek. 
If buf position is reset to 0, the first block is entirely missing. 

Thus, flushing and buf resetting should only be done when something is wrong.